README.md

```diff
@@ -1,57 +1,297 @@
+  storage_create_params = StorageModels::StorageAccountCreateParameters.new.tap do |account|
+    account.location = WEST_US
+    account.sku = StorageModels::Sku.new.tap do |sku|
+      sku.name = StorageModels::SkuName::PremiumLRS
+      sku.tier = StorageModels::SkuTier::Premium
+    end
+    account.kind = StorageModels::Kind::Storage
+    account.encryption = StorageModels::Encryption.new.tap do |encrypt|
+      encrypt.services = StorageModels::EncryptionServices.new.tap do |services|
+        services.blob = StorageModels::EncryptionService.new.tap do |service|
+          service.enabled = false
+        end
+      end
+    end
+  end
+  print_item storage_account = storage_client.storage_accounts.create(GROUP_NAME, storage_account_name, storage_create_params)
+```
+
+<a id="vnet"></a>
+### Create a virtual network
+
+```ruby
+puts 'Creating a virtual network for the VM'
+vnet_create_params = NetworkModels::VirtualNetwork.new.tap do |vnet|
+  vnet.location = WEST_US
+  vnet.address_space = NetworkModels::AddressSpace.new.tap do |addr_space|
+    addr_space.address_prefixes = ['10.0.0.0/16']
+  end
+  vnet.dhcp_options = NetworkModels::DhcpOptions.new.tap do |dhcp|
+    dhcp.dns_servers = ['8.8.8.8']
+  end
+  vnet.subnets = [
+      NetworkModels::Subnet.new.tap do |subnet|
+        subnet.name = 'rubySampleSubnet'
+        subnet.address_prefix = '10.0.0.0/24'
+      end
+  ]
+end
+print_item vnet = network_client.virtual_networks.create_or_update(GROUP_NAME, 'sample-ruby-vnet', vnet_create_params)
+```
+
+<a id="ipaddress"></a>
+### Create a public IP address
+
+```ruby
+puts 'Creating a public IP address for the VM'
+public_ip_params = NetworkModels::PublicIPAddress.new.tap do |ip|
+    ip.location = WEST_US
+    ip.public_ipallocation_method = NetworkModels::IPAllocationMethod::Dynamic
+    ip.dns_settings = NetworkModels::PublicIPAddressDnsSettings.new.tap do |dns|
+        dns.domain_name_label = 'msi-vm-domain-name-label'
+    end
+end
+print_item public_ip = network_client.public_ipaddresses.create_or_update(GROUP_NAME, 'sample-ruby-pubip', public_ip_params)
+```
+
+<a id="nic"></a>
+### Create a network interface
+
+```ruby
+print_item nic = network_client.network_interfaces.create_or_update(
+    GROUP_NAME,
+    "sample-ruby-nic-#{vm_name}",
+    NetworkModels::NetworkInterface.new.tap do |interface|
+        interface.location = WEST_US
+        interface.ip_configurations = [
+            NetworkModels::NetworkInterfaceIPConfiguration.new.tap do |nic_conf|
+                nic_conf.name = "sample-ruby-nic-#{vm_name}"
+                nic_conf.private_ipallocation_method = NetworkModels::IPAllocationMethod::Dynamic
+                nic_conf.subnet = subnet
+                nic_conf.public_ipaddress = public_ip
+            end
+        ]
+    end
+)
+```
+
+<a id="vm"></a>
+### Create a virtual machine with system identity
```

Comment:
is this needed for any VM creation or is there anything special for MSI setting, maybe comment something on that?

Reply:
Assigning the identity of type `SystemAssigned` is the only part that is different from normal vm creation. I've added that information in the explanation.